include/supernode.h

```diff
@@ -84,10 +74,27 @@ class Supernode {
     bytes_size = sizeof(Supernode) + log2(n)/(log2(3)-1) * Sketch::sketchSizeof() - sizeof(char);
   }
 
-  static inline uint32_t get_size() {
+  static inline size_t get_size() {
     return bytes_size;
   }
 
+  inline size_t get_sketch_size() {
+    return sketch_size;
+  }
+
+  // return the number of sketches held in this supernode
+  int get_num_sktch() { return num_sketches; };
+
+    // get the ith sketch in the sketch array
+  inline Sketch* get_sketch(size_t i) {
```

Comment:
Is it possible to only make the const pointer public? This might require making Sketch write_binary const but that’s doable.

Reply:
I don't think that's how `const` member functions work. Afaik `const` versions of member functions are invoked only when the object is `const`. That means the `supernode` would need to be a `const` supernode, which clearly it cannot be since we are actively modifying it,

Reply:
Why do you want this anyways?

Reply:
I'm not saying the member function should be const but rather return a const pointer to a sketch object. The reason I want this is so that the caller of get_sketch can't muck with the internals of the supernode and make the data invalid for example.

Reply:
Sketch const * is really what I want. Since we aren't returning a reference to the pointer we don't need to protect the pointer with const.

Reply:
Yeah... could also change the name of the public function. Maybe `read_sketch()` or something. I'll leave this up to you.